packages/lexical-react/src/shared/useCharacterLimit.ts

```diff
@@ -173,6 +173,9 @@ function $wrapOverflowedNodes(offset: number): void {
 
   for (let i = 0; i < dfsNodesLength; i += 1) {
     const {node} = dfsNodes[i];
+    const hasOverflowParent = !!node
+      .getParents()
+      .find((p) => p instanceof OverflowNode);
```

Comment:
Probably makes sense to compute the full condition here since $isLeafNode is cheap and traversing the parents is not so cheap.

```suggestion
    const needsOverflowParent = $isLeafNode(node) && !$findMatchingParent(node, $isOverflowNode);
```

Reply:
Good point! I changed the condition as you suggested.